src/boundary.rs

```diff
@@ -29,45 +29,58 @@ pub struct BoundaryWrap;
 pub struct BoundaryRemoval;
 
 fn boundary_wrap_system(
-    window: Res<WindowDescriptor>,
+    mut windows: ResMut<Windows>,
     mut query: Query<(&mut Transform, &Bounding), With<BoundaryWrap>>,
 ) {
-    for (mut transform, radius) in query.iter_mut() {
-        let x = transform.translation.x;
-        let y = transform.translation.y;
 
-        let half_width = window.width / 2.0;
-        if x + radius.0 * 2.0 < -half_width {
-            transform.translation.x = half_width + radius.0 * 2.0;
-        } else if x - radius.0 * 2.0 > half_width {
-            transform.translation.x = -half_width - radius.0 * 2.0;
-        }
 
-        let half_height = window.height / 2.0;
-        if y + radius.0 * 2.0 < -half_height {
-            transform.translation.y = half_height + radius.0 * 2.0;
-        } else if y - radius.0 * 2.0 > half_height {
-            transform.translation.y = -half_height - radius.0 * 2.0;
+    let temp_win = windows.get_primary_mut();
+
+    if !temp_win.is_none() {
+        let window = windows.get_primary_mut().unwrap();
+        for (mut transform, radius) in query.iter_mut() {
+            let x = transform.translation.x;
+            let y = transform.translation.y;
+    
+            let half_width = window.width() / 2.0;
+            if x + radius.0 * 2.0 < -half_width {
+                transform.translation.x = half_width + radius.0 * 2.0;
+            } else if x - radius.0 * 2.0 > half_width {
+                transform.translation.x = -half_width - radius.0 * 2.0;
+            }
+    
+            let half_height = window.height() / 2.0;
+            if y + radius.0 * 2.0 < -half_height {
+                transform.translation.y = half_height + radius.0 * 2.0;
+            } else if y - radius.0 * 2.0 > half_height {
+                transform.translation.y = -half_height - radius.0 * 2.0;
+            }
         }
     }
 }
 
 fn boundary_remove_system(
-    window: Res<WindowDescriptor>,
     mut commands: Commands,
+    mut windows: ResMut<Windows>,
```

Comment:
Since this system doesn't need to mutate any window, we could use just `Res<Windows>` here, and call `windows.get_prymary()` instead of `get_primary_mut()`.